### PR TITLE
refactor: use xml decoder to correctly process PARA as newlines

### PR DESCRIPTION
**Pull request by EngiN33R**
### Motivation

Every XML `TEXT` element is processed as a separate line, which causes incorrect display for formatted text for many FLHD strings.

image

### Solution

Use `XmlDecoder` instead of struct unmarshalling to imperatively process tags and create lines based on PARA or POP occurrences.

image

### CODE REVIEW

**Comment by dd84ai** [comment]
Screenshot_20240607_112504
Works strange on Freelancer Discovery data.
Same on Vanilla data. Not a single infocard is present.

File: configs/configs_mapped/freelancer_mapped/infocard_mapped/infocard/xml.go [hunk 1]
```diff
@@ -1,34 +1,42 @@
...
+		}
+
+		switch tok := token.(type) {
+		case xml.EndElement:
+			if tok.Name.Local == "PARA" || tok.Name.Local == "POP" {
+				lines = append(lines, line)
+				line = ""
+			}
+		case xml.CharData:
+			line += string(tok)
```
Comment (dd84ai) [inline configs/configs_mapped/freelancer_mapped/infocard_mapped/infocard/xml.go:35]: I would prefer seeing string building via String Builder
```go
package main

import (
    "strings"
    "fmt"
)

func main() {
    // ZERO-VALUE:
    //
    // It's ready to use from the get-go.
    // You don't need to initialize it.
    var sb strings.Builder

    for i := 0; i < 1000; i++ {
        sb.WriteString("a")
    }

    fmt.Println(sb.String())
}
```

File: configs/configs_mapped/freelancer_mapped/infocard_mapped/infocard/xml.go [hunk 2]
```diff
@@ -1,34 +1,44 @@
...
-	err := xml.Unmarshal([]byte(prepared), &structy)
-
-	if _, ok := err.(*xml.SyntaxError); ok {
-		replaced := strings.ReplaceAll(prepared, "&", "")
-		err = xml.Unmarshal([]byte(replaced), &structy)
+func XmlToText(raw string) ([]string, error) {
+	prepared := strings.ReplaceAll(string(raw), `<?xml version="1.0" encoding="UTF-16"?>`, "")
+	decoder := xml.NewDecoder(bytes.NewBufferString(prepared))
+
+	lines := make([]string, 0)
```
Comment (dd84ai) [inline configs/configs_mapped/freelancer_mapped/infocard_mapped/infocard/xml.go:18]:
```suggestion
	lines := make([]string, 0, 10)
```
This one last minor thing to fix and i think it is good to go.
You create by default empty array.
And apperently golang has performance issues with this approach if many times adding minor amount of lines to it in such way.

With adding `10` as third value... you tell Golang to create array with initial... inner memory allocated for 10 elements (but it will be still using it as length of array 0, 1, and etc according what u need)
It optimizes just Memory allocating, saving a lot of performance by wasting a bit more RAM memory.

https://stackoverflow.com/questions/45423667/what-is-the-point-in-setting-a-slices-capacity
in a place like this it can be read
"Golang array capacity" stuff to search.

TLDR: Setting capacity will not change anything, beyond allocating better RAM memory during inner working of array working, and increasing performance twice.

**Review by dd84ai — APPROVED** [review]
